### ISSUE TITLE
Warning: array_key_exists(): The first argument should be either a string or an integer

### ISSUE THREAD

**Issue by shefik** [zikula/core#2303]
If you get the instance of Zikula View as follows:

```
$view = Zikula_View::getInstance(null);
```

or

```
$view = Zikula_View::getInstance(null, false);
```

And, if there is no module name for the function to obtain via ModUtil::getName() or $view->toplevelmodule, then this error is thrown:

```
Warning: array_key_exists(): The first argument should be either a string or an integer in /lib/legacy/Zikula/View.php on line 417
```

There needs to be a better fallback, to avoid this error. Perhaps default to an internal Zikula identifier, or throw an exception instead, if a module name always required?

Feedback requested.

**Comment by shefik** [comment]
Would this be valid to use as a default?

ModUtil:CONFIG_MODULE (equal to "ZConfig").

**Comment by craigh** [comment]
can you please provide the use-case where you might call `getInstance` and not have a module set? How did you run into this error?

**Comment by shefik** [comment]
@craigh If I had a custom file, such as example.php, and I need to boot Zikula in that file as such:

```
include 'lib/bootstrap.php';

$request = Request::createFromGlobals();
$core->init(Zikula_Core::STAGE_ALL, $request);

$container = $core->getContainer();
$entityManager = $container->get('doctrine.entitymanager');
$view = Zikula_View::getInstance(null, false);
```

I just need a default value to insert instead of a module name, because I want to be able to reference a plugin as such:

```
require_once $view->_get_plugin_filepath('insert', 'getstatusmsg');
```

At the end of the script, I will echo the output:

```
echo '<div>'.smarty_insert_getstatusmsg(array(), $view).'</div>';
```

In this scenario, I am not accessing Zikula normally via index.php, since I don't want to load the theme and everything else Zikula would need to load via index.php. I just need to access the Zikula boot through the code indicated above.

**Comment by craigh** [comment]
I'm not sure using a second front controller is supported. I'm not sure how symfony works with that. Additionally, a module is probably always required. I guess you could look and see what happens in the core when an instance is created for the theme (`Zikula_View_Theme`) but not sure.

**Comment by Guite** [comment]
Fixed in 38008da12c76605cff05eb0fc7f36671f630cefd

**Comment by Guite** [comment]
@craigh while this is indeed not a "supported" use case in a way that we invest much effort into providing convenience, the core should still not fail with an error; so introducing a fallback seems fine here. At the end it is not really important because this is a rare usecase and Smarty is going to disappear in future.
